codegenerator/cli/templates/static/codegen/src/eventFetching/PartitionedFetchState.res

```diff
@@ -1,13 +1,14 @@
 open Belt
+type partitionIndex = int
 type t = {
   maxAddrInPartition: int,
-  partitions: list<FetchState.t>,
+  newestPartitionIndex: partitionIndex,
+  partitions: dict<FetchState.t>,
```

Comment:
Data structure changed as described in PR description.

I needed immutable operations with an int key so it might be sensible to use a belt map, but I thought we could rather stay closer to native js and just use a dict. I therefore am converting the key back and forth to string and also added a util for dict spreading.